src/spark_manager.py

```diff
@@ -143,16 +151,229 @@ def create_cluster(self,
         # Create or validate postgres config
         postgres_env = self._prepare_postgres_config()
 
-        # TODO: Create the Spark master deployment and service
+        # Create the Spark master deployment and service
+        self._create_master_deployment(master_cores, master_memory, postgres_env)
+        self._create_master_service()
+
         # TODO: Create the Spark worker deployment
 
         # Return cluster information
-        master_url = f"spark://{self.master_name}.{self.namespace}:{self.DEFAULT_MASTER_PORT}"
+        master_url = (
+            f"spark://{self.master_name}.{self.namespace}:{self.DEFAULT_MASTER_PORT}"
+        )
         master_ui_url = f"http://{self.master_name}.{self.namespace}:{self.DEFAULT_MASTER_WEBUI_PORT}"
 
         return {
             "cluster_id": self.cluster_id,
             "master_url": master_url,
             "master_ui_url": master_ui_url,
-            "status": "creating"
+            "status": "creating",
         }
+
+    def _create_master_deployment(
+        self, cores: int, memory: str, postgres_env: List[Dict[str, str]]
+    ):
+        """
+        Create the Spark master deployment.
+
+        Args:
+            cores: Number of CPU cores for the master
+            memory: Memory allocation for the master
+            postgres_env: Postgres environment variables
+        """
+        # Define Spark master environment variables
+        env = [
+            {"name": "SPARK_MODE", "value": "master"},
+            {"name": "SPARK_MASTER_PORT", "value": str(self.DEFAULT_MASTER_PORT)},
+            {
+                "name": "SPARK_MASTER_WEBUI_PORT",
+                "value": str(self.DEFAULT_MASTER_WEBUI_PORT),
+            },
```

Comment:
If we start up multiple spark masters with the same ports won't they collide?

Reply:
it won't - each of the services are within the its own pod, then port was used by the service deployment (each service has its own name).

Reply:
Ok, so the ports are namespaced by the pods, unlike rancher 1, right?

Reply:
well... technically ports are managed/namespaced by service - a pod can open multiple ports, but you need to use the service to connect to the port. For example, in our case, the master pod opens 7077 and 8090. I deployed a Service to enable other Pods to connect to those ports. I can also deploy 2 services instead of 1, so that each service manages only 1 port.

Reply:
So the service ports need to be unique?

Reply:
Within the service, ports need to be unique. 2 different services can have the same port, e.g. service_1:7077 is connecting to the 7077 port from user1/pod1 master and service_2:7077 is connecting to the 7077 port for user2/pod2 master.